### ISSUE TITLE
Swift build always fetches packages first

### ISSUE THREAD

**Issue by koraykoska** [swift-server/vscode-swift#593]
**Describe the bug**
In a project with multiple `Package.swift` files, e.g.: monorepos with multiple packages, the task `Build Debug Run` always does a full `swift package resolve`, which leads to very high build iteration times.

**To Reproduce**
1. Create a directory with multiple Vapor projects (only reproducible if the packages have multiple products, one of which at least is an executable)
2. Debug Build both projects
3. Change one line in one file and build again

You should see that a package resolve happens again, even though it should be cached.

**Expected behavior**
Not resolving packages, especially if they are fresh. I am sure there is a mechanism behind caching and at which point a new resolve should happen, but it's every single time right now.

**Environment**
 - OS: macOS, but with VSCode devcontainers in a Ubuntu 22.04 Docker Container with Swift
 - Swift version (output of `swift --version`): 5.8.1
 - Visual Studio Code version: 1.81.1
 - vscode-swift version: 1.5.0

**Comment by adam-fowler** [comment]
I have seen this, but it doesn't seem to happen 100%
If you are using devcontainers could you try a devcontainer based off the latest development build of swift 5.9. There is a fix in 5.9 which might fix this issue for you. If you could verify it, that would be great.

**Comment by koraykoska** [comment]
@adam-fowler I can try but I won't be able to switch for everyday development before 5.9 is released as the whole setup depends on 5.8.
Is there no way to fix this temporarily, even if it's a hack?

I'll try 5.9 and let you know if it makes a difference.

**Comment by adam-fowler** [comment]
If you run `swift package update` while VSCode isn't running I imagine that will fix the issue. 

But it would be really helpful if you tried 5.9 beforehand to see if that resolved the issue.

**Comment by koraykoska** [comment]
@adam-fowler Swift 5.9 did not fix it. Not even with 1.6.0 of vscode-swift.

**Comment by adam-fowler** [comment]
Ok so the fix in 5.9 didn't resolve your issue. Hmmm

I think I'm going to need a little more info. You say you have multiple Package.swift files. Can you provide your exact project structure. eg
```
Package1
  Package.swift
  ...
Package2
  Package.swift
  ...
```

One other thing to test. If you go to the settings and disable sourcekit-lsp do you still get the same issue?

**Comment by adam-fowler** [comment]
Closing as I've had no response
